public/js/event-views.js

```diff
@@ -327,6 +331,103 @@ views.DialogView = Backbone.Marionette.Layout.extend({
                 break;
         }
     },
+
+    submitContactInfo: function(event) {
+        event.preventDefault(); 
+        
+        var preferredContact = {}; //Making an object for preferred contact 
+
+        var noShareVal = false;
+        var scope = $("#my-contact-info-modal");
+        var emailInfo = $("[name=email_info]", scope).val();
+        var twitterHandle = $("[name=twitter_handle]", scope).val();
+        var linkedinURL = $("[name=linkedin_url]", scope).val();
+        var noShareChkBox = $("[name=no-share]", scope).is(':checked');
+
+        //Hiding the error classes
+        $(".empty-contact-info-error", scope).hide();
+
+        $(".email-validate-error").hide();
+        $(".twitter-validate-error").hide();
+        $(".linkedin-validate-error").hide();
+
+        $('#email_info', scope).removeClass('contact_invalid_error');
+        $('#twitter_handle', scope).removeClass('contact_invalid_error');
+        $('#linkedin_url', scope).removeClass('contact_invalid_error');
+
+        if((emailInfo === "" && twitterHandle === "" && linkedinURL === "") && 
+            noShareChkBox === false) {
+            $(".empty-contact-info-error", scope).show();
+            scope.modal('show');
+            return;
+        }
+
+        //Regular expression test for email input
+        var emailRegX = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+        
+        if(emailInfo) {
+
+            var isEmailValid = emailInfo.match(emailRegX);
+        
+            if(isEmailValid === null) {
+                $('#email_info', scope).addClass('contact_invalid_error');
+                scope.modal('show');
+                $(".email-validate-error").show();
+                return;
+            }
+        }
+
+        //Regular expression test for twitter input 
+        var twitterRegex = /^@?(\w+)$/;
```

Comment:
Let's create a new utility library that can be shared by server and client that contains all of these regexes, so that we can enforce on server as well.